internal/server/receiver_handlers.go

```diff
@@ -367,6 +369,30 @@ func (s *ReceiverServer) annotate(ctx context.Context, resource apiv1.CrossNames
 
 	group, version := getGroupVersion(apiVersion)
 
+	if resource.Name == "*" && resource.MatchLabels != nil {
```

Comment:
```suggestion
	if resource.Name == “” && resource.MatchLabels != nil {
```

I don’t think we need to support wildcard in the name but the name should take precedence over matchLabels if both are set

>

Reply:
I agree with that, it's done like that to keep things consistent how things are done for `Alert`  https://github.com/fluxcd/notification-controller/blob/main/internal/server/event_handlers.go#L293